packages/components/src/dropdown/index.js

```diff
@@ -49,10 +49,14 @@ class Dropdown extends Component {
 	 * Closes the dropdown if a focus leaves the dropdown wrapper. This is
 	 * intentionally distinct from `onClose` since focus loss from the popover
 	 * is expected to occur when using the Dropdown's toggle button, in which
-	 * case the correct behavior is to keep the dropdown closed.
+	 * case the correct behavior is to keep the dropdown closed. The same applies
+	 * in case when focus is moved to the modal dialog.
 	 */
 	closeIfFocusOutside() {
-		if ( ! this.containerRef.current.contains( document.activeElement ) ) {
+		if (
+			! this.containerRef.current.contains( document.activeElement ) &&
+			! document.activeElement.closest( '[role="dialog"]' )
```

Comment:
I have no idea whether we can assume that focus can be moved to the element with `[role="dialog"]` in all cases. Thoughts?

Reply:
It definitely feels like there needs to be a more general way to track that a modal or popover was opened from the context of another modal or popover, and exclude it from the focus outside check.

There's a similar issue here with the inline link popover, where it needs to check whether focus is in the autocomplete list to avoid closing:
https://github.com/WordPress/gutenberg/blob/fb910e55e021b66be8332fb979536d885e87ab95/packages/format-library/src/link/inline.js#L168-L179

Having said that, I can't think of a case where this logic fails, so I think it'd be pragmatic to ship this, and then consider what a more general solution might be.